MeshGeoToolsLib/MeshNodeSearcher.h

```diff
@@ -142,9 +148,9 @@ class MeshNodeSearcher
     MeshGeoToolsLib::SearchLength _search_length_algorithm;
     bool _search_all_nodes;
     // with newer compiler we can omit to use a pointer here
-    std::vector<MeshNodesOnPoint*> _mesh_nodes_on_points;
-    std::vector<MeshNodesAlongPolyline*> _mesh_nodes_along_polylines;
-    std::vector<MeshNodesAlongSurface*> _mesh_nodes_along_surfaces;
+    mutable std::vector<MeshNodesOnPoint*> _mesh_nodes_on_points;
+    mutable std::vector<MeshNodesAlongPolyline*> _mesh_nodes_along_polylines;
+    mutable std::vector<MeshNodesAlongSurface*> _mesh_nodes_along_surfaces;
```

Comment:
`_mesh_nodes_on_points`, `_mesh_nodes_along_polylines` and `_mesh_nodes_along_surfaces` are used as caches. These caches are the only internals that are modified by some methods. Until now, this fact has prevented that some of the methods are `const`.